[docs] Add note about accessing Azure OpenAI through a proxy (#186436)
elastic/kibana: docs/management/connectors/action-types/openai.asciidoc

image::management/connectors/images/gen-ai-connector.png[OpenAI connector]
// NOTE: This is an autogenerated screenshot. Do not edit it directly.

[IMPORTANT]
====
Elastic provides no official support for connecting to the Azure OpenAI service through a proxy.
However if you must use a proxy,
ensure that the proxy supports streaming and is SSE-compatible,

> or the connector will timeout when it attempts to communicate with the Azure OpenAI API.

Reviewer: streaming is indeed helpful for preventing timeouts, but more importantly, we just fail to parse the response if it is not a streamed response (which is newline-delimited JSON)

> or the connector will timeout when it attempts to communicate with the Azure OpenAI API.

Author: ```suggestion
or the connector will timeout when it attempts to communicate with the Azure OpenAI API,
and the response will not be parsed.
```

image::management/connectors/images/gen-ai-connector.png[OpenAI connector]
// NOTE: This is an autogenerated screenshot. Do not edit it directly.

[IMPORTANT]
====
Elastic provides no official support for connecting to the Azure OpenAI service through a proxy.
However if you must use a proxy,
ensure that the proxy supports streaming and is SSE-compatible,
or the connector will timeout when it attempts to communicate with the Azure OpenAI API,

> and the response will not be parsed.

Author: @dgieselaar Does this address your concern?

Reviewer: The risk of a timeout is one of the reasons why we use streaming, and it's not guaranteed to timeout. The main issue is that we will not parse a non-streamed response - I think we should focus on that, and not the timeout. Docs are here: https://platform.openai.com/docs/api-reference/chat/streaming.

Author: Got it. Thanks for the clarification!

> ensure that the proxy supports streaming and is SSE-compatible,
> or the connector will timeout when it attempts to communicate with the Azure OpenAI API,
> and the response will not be parsed.

Author: ```suggestion
ensure that the proxy supports streaming and is SSE-compatible.
Elastic will only parse streamed responses.
```
